LauncherWindows.bat

```diff
@@ -0,0 +1,66 @@
+@echo off
+echo Running Launcher.bat
+echo [Dependencies] Looking for dependencies
+setlocal enabledelayedexpansion
+
+
+set nodejs=node
+set rust=cargo
+set gitwd=git
+
+where %nodejs% >nul 2>nul
+if %errorlevel% equ 0 (
+    echo [Dependencies] %nodejs% is installed.
+) else (
+    echo [Dependencies] Node.js Not installed
+    echo [Dependencies] Installing Node.js.
+    msiexec.exe /i resources\node-v20.11.0-x64.msi /L*V resources\rustinstall.log
+)
+where %rust% >nul 2>nul
+if %errorlevel% equ 0 (
+    echo [Dependencies] %rust% is installed.
+) else (
+    echo [Dependencies] Rust not installed.
+    echo [Dependencies] Installing Rust.
+    start /wait resources\rustup-init.exe
+)
+
+where %gitwd% >nul 2>nul
+if %errorlevel% equ 0 (
+    echo [Dependencies] %gitwd% is installed.
+) else (
+    echo [Dependencies] Git not installed.
+    echo [Dependencies] Installing Git.
+    start /wait resources\Git-2.43.0-64-bit.exe
+)
+
+
+echo [Launcher] Loading include_ui
+call client-backend\include_ui.bat
+
+echo [Updater] Checking for updates
+git status | find "Your branch is up to date with 'origin/main'." > nul
```

Comment:
From my understanding `git fetch` pulls from the repository and over writes the current data every time its called but `git status` just gets the status of the the 2 repository's and compares them. So I'm just see if its out of date and then if its out of date it dose `git pull origin main` so it pull from the main branch and updates everything though I'm currently working on a way to update the launcher file because it cant update its self. And if you'd like i can change it so it just redirects them to the website if something is missing.

Reply:
`git fetch` just checks the upstream for any changes, it doesn't modify anything on the local copy. `git status` tells you the current state of your local copy (i.e. any modified files that need to commit and how many commits ahead / behind your current branch is) compared to *the last time the upstream was checked*, but does not actually check for new information from upstream. So if something has changed upstream but nothing has updated it on the local machine (e.g. a fetch or pull), then status will be outdated and continue saying you are up to date.

Reply:
ok well I'm going to put it in for redundancy